src/woollypully/DESIGN.md

```diff
@@ -0,0 +1,69 @@
+# woollypully
+
+Woollypully uses Sweater Comb rules to implement API governance for Kubernetes deployments.
+
+## Why?
+
+Prevent releases that do not conform to API standards from seeing the light of day.
+
+## Just-In-Time Gating
+
+### API Discovery
+
+Using a set of well-known endpoints on each microservice, we can:
+
+- Discover what APIs are provided (a service may provide multiple)
+- Match on those we need to govern (public OpenAPI ones)
+- Apply checks to all the OpenAPI versions provided by those
+
+The main entrypoint in this woollypully sidecar makes use of these endpoints to coordinate the gating.
+
+### Comparing current and proposed
+
+TODO: open problem...
+
+The API discovery isn't exposed outside the pod, so how would we compare with the current running service?
```

Comment:
VU possibly

Reply:
It shouldn't be difficult to expose the API discovery if needed.